packages/ethereum-contracts/contracts/agreements/CFAv1Forwarder.sol

```diff
@@ -0,0 +1,493 @@
+
+        if (flowrate > 0) {
+            if (prevFlowRate == 0) {
+                // create flow
+                cfaCallData = sender == msg.sender ?
+                    abi.encodeCall(
+                        _cfa.createFlow,
+                        (
+                            token,
+                            receiver,
+                            flowrate,
+                            new bytes(0) // placeholder
+                        )
+                    ) :
+                    abi.encodeCall(
+                        _cfa.createFlowByOperator,
+                        (
+                            token,
+                            sender,
+                            receiver,
+                            flowrate,
+                            new bytes(0) // placeholder
+                        )
+                    );
+            } else if (prevFlowRate != flowrate) {
+                // update flow
+                cfaCallData = cfaCallData = sender == msg.sender ?
+                    abi.encodeCall(
+                        _cfa.updateFlow,
+                        (
+                            token,
+                            receiver,
+                            flowrate,
+                            new bytes(0) // placeholder
+                        )
+                    ) :
+                    abi.encodeCall(
+                        _cfa.updateFlowByOperator,
+                        (
+                            token,
+                            sender,
+                            receiver,
+                            flowrate,
+                            new bytes(0) // placeholder
+                        )
+                    );
+            } else {
+                // no change
+                return;
+            }
+        } else if (flowrate == 0) {
+            if (prevFlowRate > 0) {
+                // delete flow
+                cfaCallData = sender == msg.sender ?
+                    abi.encodeCall(
+                        _cfa.deleteFlow,
+                        (
+                            token,
+                            sender,
+                            receiver,
+                            new bytes(0) // placeholder
+                        )
+                    ) :
+                    abi.encodeCall(
+                        _cfa.deleteFlowByOperator,
+                        (
+                            token,
+                            sender,
+                            receiver,
+                            new bytes(0) // placeholder
+                        )
+                    );
+            } else {
+                // do nothing
+                return;
+            }
+        } else {
+            revert("invalid flowrate");
+        }
+
+        _forwardBatchCall(address(_cfa), cfaCallData);
+    }
+
+    function _createFlow(ISuperToken token, address sender, address receiver, int96 flowrate, bytes memory userData)
```

Comment:
I think we can and should reuse these functions below: `_createFlow`, `_updateFlow`, `_deleteFlow` above in the `_setFlowrateFrom` function - there is some duplication. The only difference I am noticing is that it takes a userData which will just be `new bytes(0)` in this case.

Reply:
oh damn, missed that possible simplification. Love you!